stories/src/table/with-dimension-locking.jsx

```diff
@@ -0,0 +1,300 @@
+// @flow
+import React, { Component, Fragment } from 'react';
+import type { Node } from 'react';
+import styled from 'styled-components';
+import { DragDropContext, Droppable, Draggable } from '../../../src';
+import reorder from '../reorder';
+import { colors, grid } from '../constants';
+import type { Quote } from '../types';
+import type {
+  DropResult,
+  DroppableProvided,
+  DraggableProvided,
+  DraggableStateSnapshot,
+} from '../../../src';
+
+const Table = styled.table`
+  width: 500px;
+  margin: 0 auto;
+  table-layout: ${props => props.layout};
+`;
+
+const TBody = styled.tbody`
+  border: 0;
+`;
+
+const THead = styled.thead`
+  border: 0;
+  border-bottom: none;
+  background-color: ${colors.grey.light};
+`;
+
+const Row = styled.tr`
+  ${props => (props.isDragging ? `background: ${colors.green};` : '')}
+`;
+
+const Cell = styled.td`
+  box-sizing: border-box;
+  padding: ${grid}px;
+`;
+
+type TableCellProps = {|
+  children: Node,
+  isDragOccurring: boolean,
+|}
+
+type TableCellSnapshot = {|
+  width: number,
+  height: number,
+|}
+class TableCell extends React.Component<TableCellProps> {
+  // eslint-disable-next-line react/sort-comp
+  ref: ?HTMLElement
```

Comment:
Can use [ElementRef](https://flow.org/en/docs/react/types/#toc-react-elementref) to type these variables.

Reply:
These are actualy HTMLElement and not component instances